lib/ssl/src/ssl_handshake.erl

```diff
@@ -745,6 +754,38 @@ encode_extensions([#pre_shared_key_server_hello{selected_identity = Identity} |
     encode_extensions(Rest, <<?UINT16(?PRE_SHARED_KEY_EXT),
                               ?UINT16(2), ?UINT16(Identity), Acc/binary>>).
 
+encode_cert_status_req(
+    StatusType,
+    #ocsp_status_request{
+        responder_id_list = ResponderIDList,
+        request_extensions = ReqExtns}) ->
+    ResponderIDListBin = encode_responderID_list(ResponderIDList),
+    ReqExtnsBin = encode_request_extensions(ReqExtns),
+    <<?BYTE(StatusType), ResponderIDListBin/binary, ReqExtnsBin/binary>>.
+
+encode_responderID_list([]) ->
+    <<?UINT16(0)>>;
+encode_responderID_list(List) ->
+    do_encode_responderID_list(List, <<>>).
+
+%% ResponderID is DER-encoded ASN.1 type defined in RFC6960.
+do_encode_responderID_list([Responder | Rest], Acc)
+  when is_binary(Responder) ->
+    Len = byte_size(Responder),
+    do_encode_responderID_list(
+        Rest, <<Acc/binary, ?UINT16(Len), Responder/binary>>);
+do_encode_responderID_list([], Acc) ->
```

Comment:
This could be the first clause instead to follow to pattern.

Reply:
> This could be the first clause instead to follow to pattern.

Done, but shouldn't we put the match clause with highest possibility to the first?

Reply:
I think such optimizations shall be done by the compiler or JIT (that is worked on). Unless it
is measured to be a bottleneck, which is seldom the case.